lib/cli/src/commands/package/download.rs

```diff
@@ -93,18 +93,48 @@ impl PackageDownload {
 
         step_num += 1;
 
-        let (full_name, version, api_endpoint, token) = match &self.package {
+        let (download_url, token) = match &self.package {
             PackageSpecifier::Registry { full_name, version } => {
                 let endpoint = self.env.registry_endpoint()?;
                 let version = version.to_string();
                 let version = if version == "*" { None } else { Some(version) };
+                let token = self.env.get_token_opt().map(|x| x.to_string());
 
-                (
-                    full_name,
-                    version,
-                    endpoint,
-                    self.env.get_token_opt().map(|x| x.to_string()),
+                let package = wasmer_registry::query_package_from_registry(
+                    endpoint.as_str(),
+                    &full_name,
+                    version.as_deref(),
+                    token.as_deref(),
                 )
+                .with_context(|| {
+                    format!(
+                "could not retrieve package information for package '{}' from registry '{}'",
+                full_name, endpoint,
+            )
+                })?;
+
+                let download_url = package
+                    .pirita_url
+                    .context("registry does provide a container download container download URL")?;
```

Comment:
typo: `"registry '{}' does not provide a container download URL", endpoint`, maybe?